File: Assignment_30/question5.cpp
////////////////////////////////////////////////////////////////////////////
//
//  File Name   : question5.cpp
//  Description : Check whether 1st and 32nd bits are ON or OFF.
//  Problem     : Write a program which checks whether the 1st and 32nd
//                bits in the given number are ON or OFF.
//
//  Author      : Asawari Jadhav
//  Date        : 18/06/2025
//
////////////////////////////////////////////////////////////////////////////

#include <iostream>
using namespace std;

typedef unsigned int UINT;

bool CheckBit(UINT iNo)
{
  UINT iMask = 0x80000001; // 1st, 32nd  bits
  UINT iResult = 0;

  iResult = iNo & iMask;

  return (iResult == iMask);
}

int main()
{
  UINT iValue = 0;
  bool bRet = false;

  cout << "Enter number : \n";
  cin >> iValue;

  bRet = CheckBit(iValue);

  if (bRet == true)
  {
    cout << "1st and 32nd bits are ON\n";
  }
  else
  {
    cout << "One or both bits are OFF\n";
  }

  return 0;
}

File: Assignment_45/question5.java

//////////////////////////////////////////////////////////////////////////////////////////////
//
//  
//    File Name    :  question5.java
//    Description  :  Check whether matrix is a sparse matrix
//    Problem      :  Sparse matrix: majority elements are 0
//
//    Author       :  Asawari Jadhav
//
//    Date         :  30-07-2025
//  
//  
//
//////////////////////////////////////////////////////////////////////////////////////////////

  class MatrixX
{
    public boolean ChkSparse(int Arr[][], int iRow, int iCol)
    {
        int zeroCount = 0;
        int totalElements = iRow * iCol;

        for(int i = 0; i < iRow; i++)
        {
            for(int j = 0; j < iCol; j++)
            {
                if(Arr[i][j] == 0)
                    zeroCount++;
            }
        }

        return (zeroCount > totalElements / 2);
    }
}

public class question5
{
    public static void main(String args[])
    {
        int[][] Arr = {
            {1, 0, 0},
            {0, 0, 0},
            {9, 0, 0}
        };

        MatrixX mobj = new MatrixX();
        boolean bRet = mobj.ChkSparse(Arr, 3, 3);

        if(bRet)
        {
            System.out.println("Matrix is Sparse Matrix");
        }
            
        else
        {
             System.out.println("Matrix is not Sparse Matrix");
        }
           
    }
}
